src/Equinor.ProCoSys.IPO.WebApi/Synchronization/StringToGuidConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Equinor.ProCoSys.IPO.WebApi.Synchronization;
+
+public class StringToGuidConverter : JsonConverter<Guid>
+{
+    public override Guid Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.String)
+        {
+            var stringGuid = reader.GetString();
+
+            // Try parsing with hyphens (D format).
+            if (Guid.TryParseExact(stringGuid, "D", out Guid guid))
+            {
+                return guid;
+            }
+
+            // If the D format fails, try without hyphens (N format).
+            else if (Guid.TryParseExact(stringGuid, "N", out guid))
+            {
+                return guid;
+            }
+
+            throw new JsonException($"Unable to deserialize JSON. Cannot convert {stringGuid} to {typeof(Guid)}.");
+        }
+
+        throw new JsonException("The JSON value could not be read as a guid.");
```

Comment:
Is this error message correct? I read it as if TokenType is not JsonTokenType.String then throw exception.
So error message should be something like "StringToGuidConverter: JsonTokenType is not of type String, unable to convert to guid" or something along those lines.

Reply:
Changing message to: "The JSON value is not of type string and can not be read as a guid."